debug/org.eclipse.debug.tests/src/org/eclipse/debug/tests/ui/LaunchConfigurationTabGroupViewerTest.java

```diff
@@ -71,7 +69,7 @@ public void testAllTabsAreInitializedByDefault() {
 		final ILaunchConfigurationTab[] tabs = runOnDialog(createAndSelect1LaunchConfig);
 
 		for (int i = 0; i < tabs.length; i++) {
-			assertThat(((SpyTab) tabs[i])).withFailMessage("tab %s was not initialized", i).matches(SpyTab::isInitialized);
+			assertThat(((SpyTab) tabs[i]).isInitialized()).withFailMessage("tab %s was not initialized", i).isTrue();
```

Comment:
Why not keep using `matches`?
```suggestion
			assertThat((SpyTab) tabs[i]).matches(SpyTab::isInitialized, "is initialized");
```

Reply:
I find `isTrue` / `isFalse` more intuitive

Reply:
I would argue against that but that's probably rather a matter of personal preference.
Drawback of manual boolean checks with `withFailMessage` is that you have to manually ensure that the object used for creating the failure message matches the object on which the assertion is defined. This is implicitly ensured when using `matches`.

Reply:
You mean the 2nd parameter  (the `i`) in the call to `withFailMessage`?

Reply:
I mean the subject under test (i.e., the object passed to `assertThat`). In this case, the test class replicates logic for creating a descriptive name for the subject under test, but it would be better to simply define a proper `toString()` representation for `SpyTab`.